client/src/store/slices/filterSlice/filterSlice.js

```diff
@@ -0,0 +1,66 @@
+import { createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+  tours: [],
+  prices: [0, 0],
+  duration: [],
+  seasons: [],
+  isFilter: false,
+};
+
+const filterSlice = createSlice({
+  name: 'filter',
+  initialState,
+
+  reducers: {
+    setFilteredTours: (state, action) => {
+      state.tours = action.payload;
+    },
+    setPrices: (state, action) => {
+      state.prices = action.payload;
+    },
+    setIsFilter: (state, action) => {
+      state.isFilter = action.payload;
+    },
+    setDuration: (state, action) => {
+      if (state.duration.includes(action.payload)) {
+        state.duration = state.duration.filter((el) => el !== action.payload);
+      } else {
+        state.duration.push(action.payload);
+      }
+    },
+    setClearDuration: (state, action) => {
+      state.duration = action.payload;
+    },
+    setSeasons: (state, action) => {
+      if (state.seasons.includes(action.payload)) {
+        state.seasons = state.seasons.filter((el) => el !== action.payload);
+      } else {
+        state.seasons.push(action.payload);
+      }
+    },
+    setAllSeasons: (state, action) => {
+      state.seasons = action.payload;
+    },
+  },
+});
+
+export const { setFilteredTours, setPrices, setIsFilter, setDuration, setSeasons, setAllSeasons, setClearDuration } =
+  filterSlice.actions;
+
+
+  export const fetchFilteredTours = (params) => async (dispatch) => {
+    try {
+      const { data, status } = await axios(`/products/filter?${params}`);
+  
+      if (status) {
```

Comment:
```suggestion
      if (status === 200) {
```

as any status returned even if 400 or 401 will be translated to true, so be more specific and specify the 200 status which is the success status we need.

Reply:
done